Corona-Warn-App/src/main/java/de/rki/coronawarnapp/task/TaskFactory.kt

```diff
@@ -0,0 +1,20 @@
+package de.rki.coronawarnapp.task
+
+interface TaskFactory<
+    ProgressType : Task.Progress,
+    ResultType : Task.Result
+    > {
+
+    interface Config {
+        val collisionBehavior: CollisionBehavior
+
+        enum class CollisionBehavior {
```

Comment:
I'd prefer this not to be inlined in the Config.

Reply:
It's specific to the task factories configuration. It's a value that classes that implement `interface TaskFactory.Config` need to provide, i think it's clean that the interface would also be the location that defines the `enum`.

If it would be more globally used, I might agree here, but this should only ever be declared by the task factory, and read by the `TaskController`.

I don't think extracting this increases readability or code understanding. If you don't feel strongly about this, I'd keep it here. Let's see how the others feel, I don't feel too strongly about it either.